internal/services/cdn/cdn_frontdoor_firewall_policy_resource.go

```diff
@@ -0,0 +1,1095 @@
+	result := make([]frontdoor.ManagedRuleOverride, 0)
+	if len(input) == 0 {
+		return nil
+	}
+
+	for _, v := range input {
+		rule := v.(map[string]interface{})
+
+		enabled := frontdoor.ManagedRuleEnabledStateDisabled
+		if rule["enabled"].(bool) {
+			enabled = frontdoor.ManagedRuleEnabledStateEnabled
+		}
+		ruleId := rule["rule_id"].(string)
+		action := frontdoor.ActionType(rule["action"].(string))
+		exclusions := expandCdnFrontDoorFirewallManagedRuleGroupExclusion(rule["exclusion"].([]interface{}))
+
+		result = append(result, frontdoor.ManagedRuleOverride{
+			RuleID:       &ruleId,
+			EnabledState: enabled,
+			Action:       action,
+			Exclusions:   exclusions,
+		})
+	}
+
+	return &result
+}
+
+func flattenCdnFrontDoorFirewallCustomRules(input *frontdoor.CustomRuleList) []interface{} {
+	if input == nil || input.Rules == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+	for _, v := range *input.Rules {
+		action := ""
+		if v.Action != "" {
+			action = string(v.Action)
+		}
+
+		enabled := false
+		if v.EnabledState != "" {
+			enabled = v.EnabledState == frontdoor.CustomRuleEnabledStateEnabled
+		}
+
+		name := ""
+		if v.Name != nil {
+			name = *v.Name
+		}
+
+		priority := 0
+		if v.Priority != nil {
+			priority = int(*v.Priority)
+		}
+
+		rateLimitDurationInMinutes := 0
+		if v.RateLimitDurationInMinutes != nil {
+			rateLimitDurationInMinutes = int(*v.RateLimitDurationInMinutes)
+		}
+
+		rateLimitThreshold := 0
+		if v.RateLimitThreshold != nil {
+			rateLimitThreshold = int(*v.RateLimitThreshold)
+		}
+
+		ruleType := ""
+		if v.RuleType != "" {
+			ruleType = string(v.RuleType)
+		}
+
+		results = append(results, map[string]interface{}{
+			"action":                         action,
+			"enabled":                        enabled,
+			"match_condition":                flattenCdnFrontDoorFirewallMatchConditions(v.MatchConditions),
+			"rate_limit_duration_in_minutes": rateLimitDurationInMinutes,
+			"rate_limit_threshold":           rateLimitThreshold,
+			"priority":                       priority,
+			"name":                           name,
+			"type":                           ruleType,
+		})
+	}
+
+	return results
+}
+
+func flattenCdnFrontDoorFirewallMatchConditions(input *[]frontdoor.MatchCondition) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+	for _, v := range *input {
+		selector := ""
+		if v.Selector != nil {
+			selector = *v.Selector
+		}
+
+		negateCondition := false
+		if v.NegateCondition != nil {
+			negateCondition = *v.NegateCondition
+		}
+
+		results = append(results, map[string]interface{}{
+			"match_variable":     string(v.MatchVariable),
+			"match_values":       v.MatchValue,
+			"negation_condition": negateCondition, // TODO: why isn't this negate_condition in the schema? WS: For consistency with all of the other existing firewall policies(e.g. web application and legacy frontdoor).
```

Comment:
Remove comments/TODO? Or make clear note about design decision here for future contribution understanding?

Reply:
Fixed.